### PR TITLE
[WIP] adding functionality on last seen.

### PR DESCRIPTION
**Pull request by theVJagrawal**
### Description:

#20406

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by theVJagrawal** [comment]
hey @bx80 I have added a button and the sorting function in PagedUsersList.vue file but it seems not working fine so could you please review the code and help me on this

File: plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue [hunk 1]
```diff
@@ -214,7 +214,11 @@
               :title="translate('UsersManager_UsesTwoFactorAuthentication')"
           >{{ translate('UsersManager_2FA') }}
           </th>
-          <th v-if="currentUserRole === 'superuser'">{{ translate('UsersManager_LastSeen') }}</th>
+          <th v-if="currentUserRole === 'superuser'"
+       @click="sortLastSeen">{{ translate('UsersManager_LastSeen') }}
+       <button v-if="sortedByLastSeen"
+       :class="{'fa-sort-up': sortLastSeenAsc, 'fa-sort-down': !sortLastSeenAsc}"
```
Comment (sgiehl) [inline plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue:220]: We are using our own font and icon set, so this classes may not work.

File: plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue [hunk 2]
```diff
@@ -516,6 +530,9 @@ export default defineComponent({
   },
   data(): PagedUsersListState {
     return {
+      sortLastSeenAsc: true,
+      sortedByLastSeen: false,
+      filteredUsers: [],
```
Comment (sgiehl) [inline plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue:535]: You have introduced a new `filteredUsers` object. But that one is actually not used anywhere. So you are sorting a list, that isn't displayed anywhere.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Thanks for this first approach.
Unfortunatelly it may not fully work this way. We are fetching the users that are displayed through our API paged. So if the sorting should be changed, that would need to be done through the API. Otherwise only the currently displayed users might get sorted. If there are more than 20 users this would then be incorrect.

**Comment by theVJagrawal** [comment]
hey @sgiehl thanks for reviewing the code so can you please tell me which api should i use for this sorting feature

**Comment by sgiehl** [comment]
The api call for the users is triggered here I guess:
https://github.com/matomo-org/matomo/blob/2f7c352b65624ea5d84e9cd8c06d9de90ef1d74c/plugins/UsersManager/vue/src/UsersManager/UsersManager.vue#L334-L343

Our APIs have a parameter that can be provided to sort the result. In this case that would be `filter_sort_column=last_seen`, to change the order you might need to add `filter_sort_order=asc`.

BUT: This currently doesn't work for the `UsersManager.getUsersPlusRole` API, as it only works for API methods that return a DataTable. The API could be changed to do that by returning `return DataTable::makeFromIndexedArray($users);` here:
https://github.com/matomo-org/matomo/blob/8a5c0137c74c916dd375f588d16f9b9e18fc548c/plugins/UsersManager/API.php#L421
But I'm unable to say if that change could have any side effects without looking deeper into it.

**Comment by sgiehl** [comment]
Closing this one for now. @theVJagrawal Feel free to reopen this one anytime, when you were able to finish the required changes.

**Pull request closed without merging**